Data/ByteString/Short/Internal.hs

```diff
@@ -111,6 +111,14 @@ import Prelude ( Eq(..), Ord(..), Ordering(..), Read(..), Show(..)
                , return
                , Maybe(..) )
 
+#if MIN_VERSION_base(4,15,0)
+import GHC.Exts ( word8ToWord#, wordToWord8# )
```

Comment:
Looks reasonable, but I need a confirmation that these functions will be present in`base-4.15`. @bgamari @hsyl20

Reply:
https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4492#note_315763 hows this?

Reply:
AFAICT this comment does not necessarily mean that the patch will land in GHC 9.0 and not in GHC 9.2. Could this be confirmed somehow?

Reply:
@Bodigrim Well it's a bit of an chicken and egg, because @bgamari doesn't like to merge changes to the submodule which aren't upstream yet, lest they don't get merged and GHC is stuck with a dangling reference.

Alex isn't even a submodule, so I had to really take the plunge with https://github.com/simonmar/alex/releases/tag/3.2.6, if that gives you any reassurance this isn't the most vanguard.

I wish there was a different process to only rope in library maintainers had already made the change, but that sadly doesn't exist yet.

Reply:
Current state of this repo is as follows:

* `master` branch tracks will-be-0.12 series (won't be released before 2022, I suppose). 
* `bytestring-0.11` branch tracks 0.11 series (present-day Hackage releases).
* Tag `0.10.12.0` is the latest release in 0.10 series (discontinued). 

Normally PRs land into `master`, and I backport them into `bytestring-0.11`, while 0.10 series are discontinued. Nevertheless @bgamari decided that GHC 9.0 will stick to 0.10 series, so he is responsible for rolling out whatever is needed for GHC 9.0 as `bytestring-0.10.X.X`. Please coordinate with him, if your changes are to be a part of GHC 9.0. 

Otherwise, if your changes are to make way no earlier than GHC 9.2, here is a plan:

* GHC 9.0 alpha2 (or beta, or RC) is rolled out with `unsafeWithForeignPtr`. 
* I can test and merge #333 into `master`, and backport it to `bytestring-0.11`.
* GHC switches `bytestring` submodule to track `bytestring-0.11` branch.
* You amend your commit with CPP targeting `base-4.16` and rebase it.
* I merge your commit into `master` and backport to `bytestring-0.11`. 
* GHC bumps `bytestring` submodule again.

At the moment I cannot merge neither #333, nor your changes, because they break the GHC 9.0 alpha1 build. If you change CPP to kick in starting from `base-4.16`, I can speculatively merge your commit into `master`, but it does not seem to help?..